src/common/curl_tools.c

```diff
@@ -28,6 +28,10 @@ void dt_curl_init(CURL *curl, gboolean verbose)
 {
   curl_easy_reset(curl);
 
+  char useragent[32];
```

Comment:
Indeed maybe too short, better use 64 here.

Reply:
But this (i.e. something like 4.1.0+620~g6e5bd4d3e-dirty) is almost the longest string that can be here. Except the commit count can be 4 digits long, so this string can be 1 character longer, so 27 characters. 32 seems like sufficient length.

Reply:
why have a magic constant if it can be calculated, even at runtime?

Reply:
Sure, but will this be true tomorrow. I see to possibilities:
1. Use a rather big string size to ensure any version will fit (27 is rather close to 32 and it depends on what Git output in such case). That's why I have proposed 64.
2. Compute the final size and do a dynamic allocation.

I'm not in favor of 2 for such trivial job. A 64 bytes on the stack is not a big stress!

Reply:
curl is expected to copy the UA string into its internal storage as well - it would not hurt to have 64B on stack.